File: app/src/main/java/com/walkercoding/videocompress/CompressTest.java
package com.walkercoding.videocompress;

import android.content.Context;
import android.os.AsyncTask;
import android.os.Environment;
import android.support.annotation.NonNull;
import android.util.Log;
import com.walkercoding.videocompress.videocompress.CompressParam;
import com.walkercoding.videocompress.videocompress.VideoCompressor;

import java.io.File;
import java.util.Locale;

public class CompressTest {

    //        private static final String TEST_VIDEO_FILE_PATH = "/storage/emulated/0/DCIM/Video/V80608-140857.mp4"; // 10s
    private static final String TEST_VIDEO_FILE_PATH = "/storage/emulated/0/DCIM/Video/V80209-151237.mp4"; // 1 min 122M

    public static void test(Context context) {
        File videoFile = new File(TEST_VIDEO_FILE_PATH);
        File cacheFile = new File(getExternalPictureFileDir(context), String.format(Locale.US, "tmp_%d.mp4", System.currentTimeMillis()));
        Log.e("zl", "cache File " + cacheFile.getAbsolutePath());

        final CompressParam editInfo = new CompressParam();
        editInfo.filePath = videoFile.getAbsolutePath();
        editInfo.cacheFilePath = cacheFile.getPath();

        editInfo.originalWidth = 1920;
        editInfo.originalHeight = 1080;
        editInfo.resultWidth = editInfo.originalWidth / 2;
        editInfo.resultHeight = editInfo.originalHeight / 2;
        editInfo.bitrate = 921600;

        new MyTask(editInfo).execute();
    }

    private static class MyTask extends AsyncTask<Void, Void, Void> {
        private final CompressParam editInfo;

        public MyTask(CompressParam editInfo) {
            this.editInfo = editInfo;
        }

        @Override
        protected Void doInBackground(Void... voids) {
            new VideoCompressor().convertVideo(editInfo);
            return null;
        }
    }

    public static File getExternalPictureFileDir(Context context) {
        File rootDir = new File(getExternalStorageDirectory(), context.getPackageName());
        if (rootDir.exists() && !rootDir.isDirectory()) {
            rootDir.delete();
        }

        File dir = new File(getExternalStorageDirectory(), String.format("%s/%s", context.getPackageName(), "videoPress"));
        mkdirsIfNeed(dir);
        return dir;
    }

    private static File getExternalStorageDirectory() {
        return Environment.getExternalStorageDirectory();
    }

    private static void mkdirsIfNeed(@NonNull File dir) {
        if (dir.exists() && !dir.isDirectory()) {
            dir.delete();
        }

        if (!dir.exists()) {
            dir.mkdirs();
        }
    }
}

File: app/src/main/java/com/walkercoding/videocompress/CompressTest.kt
package com.walkercoding.videocompress

import android.content.Context
import android.util.Log
import com.walkercoding.videocompress.videocompress.CompressParam
import com.walkercoding.videocompress.videocompress.VideoCompressor
import java.io.File
import java.util.*

object CompressTest {

    private class Case(val inputPath: String, val width: Int, val height: Int, val durationUs: Long)

    private val MEIZU_1 = Case("/storage/emulated/0/DCIM/Video/V80608-140857.mp4", 1080, 1920, 10 * 1000 * 1000)
    private val MEIZU_2 = Case("/storage/emulated/0/DCIM/Video/V80209-151237.mp4", 1920, 1080, 61 * 1000 * 1000)
    private val ONEPLUS_1 = Case("/storage/emulated/0/DCIM/Camera/VID_20180616_225210.mp4", 1280, 720, 74 * 1000 * 1000)

    private fun getCompressParam(context: Context, case: Case): CompressParam {
        val videoFile = File(case.inputPath)
        val cacheFile = File(Utils.getExternalPictureFileDir(context), String.format(Locale.US, "tmp_%d.mp4", System.currentTimeMillis()))

        val compressParam = CompressParam()
        compressParam.filePath = videoFile.absolutePath
        compressParam.cacheFilePath = cacheFile.path

        compressParam.originalWidth = case.width
        compressParam.originalHeight = case.height
        compressParam.durationUs = case.durationUs
        compressParam.resultWidth = compressParam.originalWidth / 2
        compressParam.resultHeight = compressParam.originalHeight / 2
        compressParam.bitrate = 921600
        compressParam.rotationValue = 270

        return compressParam
    }

    fun test(context: Context) {
        compress(getCompressParam(context, MEIZU_2))
    }

    private fun compress(param: CompressParam) {
        VideoCompressor.compress(param)
                .doOnSubscribe { Log.e("zl", "Start path ${param.filePath}") }
//                .doOnNext { Log.e("zl", "progress $it path ${param.filePath}") }
                .doOnError { Log.e("zl", "Error $it path ${param.filePath}") }
                .doOnComplete { Log.e("zl", "Done ${param.filePath}") }
                .subscribe({}, {})
    }
}
